crates/bevy_ecs/src/world/mod.rs

```diff
@@ -758,34 +765,15 @@ impl World {
     ) -> Result<[EntityMut<'_>; N], QueryEntityError> {
         Self::verify_unique_entities(&entities)?;
 
-        // SAFETY: Each entity is unique.
-        unsafe { self.get_entities_mut_unchecked(entities) }
-    }
-
-    /// # Safety
-    /// `entities` must contain no duplicate [`Entity`] IDs.
-    unsafe fn get_entities_mut_unchecked<const N: usize>(
-        &mut self,
-        entities: [Entity; N],
-    ) -> Result<[EntityMut<'_>; N], QueryEntityError> {
         let world_cell = self.as_unsafe_world_cell();
+        // SAFETY: The `world_cell` has exclusive access to the entire world,
```

Comment:
> `DeferredWorld::from(self).get_many_entities_mut();`

This doesn't work unfortunately:
```
error[E0515]: cannot return value referencing temporary value
   --> crates\bevy_ecs\src\world\mod.rs:766:9
    |
766 |         DeferredWorld::from(self).get_many_entities_mut(entities)
    |         -------------------------^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |         |
    |         returns a value referencing data owned by the current function
    |         temporary value created here
```

Reply:
Oh, I see, `&mut DeferredWorld` is only a borrow of a borrow.  To make that work, you'd need `DeferredWorld.get_many_entities_mut()` to take `self` by value so that it can return values with the `'w` lifetime, but then you'd need to call `.reborrow()` almost every time you use it.